numpy/core/src/multiarray/arrayobject.c

```diff
@@ -387,26 +474,42 @@ array_dealloc(PyArrayObject *self)
         PyObject_ClearWeakRefs((PyObject *)self);
     }
     if (fa->base) {
-        /*
-         * UPDATEIFCOPY means that base points to an
-         * array that should be updated with the contents
-         * of this array upon destruction.
-         * fa->base->flags must have been WRITEABLE
-         * (checked previously) and it was locked here
-         * thus, unlock it.
-         */
-        if (fa->flags & NPY_ARRAY_UPDATEIFCOPY) {
-            PyArray_ENABLEFLAGS(((PyArrayObject *)fa->base),
-                                                    NPY_ARRAY_WRITEABLE);
-            Py_INCREF(self); /* hold on to self in next call */
-            if (PyArray_CopyAnyInto((PyArrayObject *)fa->base, self) < 0) {
+        int retval;
+        if (PyArray_FLAGS(self) & NPY_ARRAY_WRITEBACKIFCOPY)
+        {
+            char * msg = "WRITEBACKIFCOPY requires a call to "
+                "PyArray_ResolveWritebackIfCopy before array_dealloc is "
```

Comment:
Should probably mention `DiscardWritebackIfCopy` here as another option.

Reply:
@mattip pinging this in case you missed it

Reply:
thanks